cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringServerHandler.java

```diff
@@ -194,21 +224,54 @@ public void createTethering(FullHttpRequest request, HttpResponder responder, @P
   @POST
   @Path("/tethering/connections/{peer}")
   public void tetheringAction(FullHttpRequest request, HttpResponder responder, @PathParam("peer") String peer)
-    throws NotImplementedException, BadRequestException, PeerNotFoundException, IOException {
+    throws NotImplementedException, BadRequestException, IOException {
+    checkTetheringServerEnabled();
+
     String content = request.content().toString(StandardCharsets.UTF_8);
     TetheringActionRequest tetheringActionRequest = GSON.fromJson(content, TetheringActionRequest.class);
-    TetheringStatus tetheringStatus;
+    PeerInfo peerInfo;
+    try {
+      peerInfo = store.getPeer(peer);
+    } catch (PeerNotFoundException e) {
+      responder.sendString(HttpResponseStatus.OK,
```

Comment:
similar to above, it's better if this is a 404

Reply:
fixed